src/hooks/usePermit2.ts

```diff
@@ -97,6 +120,6 @@ export default function usePermit(amount?: CurrencyAmount<Token>, spender?: stri
         return { state: PermitState.PERMITTED, signature }
       }
     }
-    return { state: PermitState.PERMIT_NEEDED, callback }
-  }, [amount, callback, isAllowed, isPermitted, isSigned, permitAllowance, signature, tokenAllowance])
+    return { state: PermitState.PERMIT_NEEDED, pending: syncState !== SyncState.SYNCED, callback }
```

Comment:
isPending?

Reply:
shouldn't this be `SyncState.PENDING`

Reply:
Left an inline comment: we want the UI to show pending unless we are synced.